src/polymer/declaration-property-handlers.ts

```diff
@@ -71,11 +72,23 @@ export function declarationPropertyHandlers(
     },
     listeners(node: estree.Node) {
       if (node.type !== 'ObjectExpression') {
+        declaration.warnings.push({
+          code: 'invalid-listeners-declaration',
+          message: '`listeners` property should be an object expression',
+          severity: Severity.ERROR,
+          sourceRange: document.sourceRangeForNode(node)!
+        });
         return;
       }
       for (let prop of node.properties) {
         if (prop.key.type !== 'Literal' || typeof prop.key.value !== 'string' ||
             prop.value.type !== 'Literal' || typeof prop.value.value !== 'string') {
+          declaration.warnings.push({
+            code: 'invalid-listeners-declaration',
+            message: 'handler value should be a string',
+            severity: Severity.ERROR,
+            sourceRange: document.sourceRangeForNode(prop.value)!
+          });
```

Comment:
Awesome!

One last change and I think this is good to go:

It's possible that `prop.key` isn't a Literal or that literal isn't a string. consider:

```javascript
const propName = 'foo';
Polymer({
  listeners: {
    [propName + '-changed']: 'handlePropChanged'
  }
});
```

As written this would give a warning on the 'handlePropName' string literal, not the computed object key, which is where the issue is. I think we'll need to split this out into two conditionals to get it right, one for the key and one for the value.